### PR TITLE
added dependency to vpa-crd for vpa-app

### PR DESCRIPTION
**Pull request by QuantumEnigmaa**
### What this PR does / why we need it

towards https://github.com/giantswarm/roadmap/issues/2009

### Checklist

- [x] Update changelog in CHANGELOG.md.

### Trigger e2e tests

/test create
/test upgrade

### CODE REVIEW

File: helm/default-apps-gcp/templates/apps.yaml [hunk 1]
```diff
@@ -6,6 +6,10 @@ kind: App
 metadata:
   annotations:
     chart-operator.giantswarm.io/force-helm-upgrade: "{{ .forceUpgrade }}"
+    {{- if and .dependsOn }}
```
Comment (calvix) [inline helm/default-apps-gcp/templates/apps.yaml:9]: `and` is used if there are two conditions that needs to be true, but i only see a one so why use it?

**Comment by calvix** [comment]
what is the value of `.dependsOn` supposed to be? and where it's set?

**Comment by QuantumEnigmaa** [comment]
My bad, this was a remnant of a previous template that I din't checked

File: helm/default-apps-gcp/templates/apps.yaml (hunk 1 above)
Comment (QuantumEnigmaa) [reply helm/default-apps-gcp/templates/apps.yaml]: My bad, this was a remnant of a previous template that I din't checked. I removed it.

**Comment by QuantumEnigmaa** [comment]
> what is the value of .dependsOn supposed to be? and where it's set?

`dependsOn` is currently only set for `vertical-pod-autoscaler-app` to allow it to wait for the vpa-crd app to be installed before installing itself. This way we avoid conflicts with CRDs creation (cf @QuentinBisson explanation in the linked issue)

It is defined in the values (sorry I also forgot to set it there at first)

**Comment by calvix** [comment]
> > what is the value of .dependsOn supposed to be? and where it's set?
> 
> `dependsOn` is currently only set for `vertical-pod-autoscaler-app` to allow it to wait for the vpa-crd app to be installed before installing itself. This way we avoid conflicts with CRDs creation (cf @QuentinBisson explanation in the linked issue)
> 
> It is defined in the values (sorry I also forgot to set it there at first)

ok

**Comment by QuantumEnigmaa** [comment]
This was tested on a CAPA cluster (golem for instance) but true, I should test it on on a CAPG one

File: helm/default-apps-gcp/templates/apps.yaml [hunk 2]
```diff
@@ -6,6 +6,10 @@ kind: App
 metadata:
   annotations:
     chart-operator.giantswarm.io/force-helm-upgrade: "{{ .forceUpgrade }}"
+    {{- if .dependsOn }}
+    # App is deployed in the org- namespace so the secret name is prefixed by the cluster-id
```
Comment (fiunchinho) [inline helm/default-apps-gcp/templates/apps.yaml:10]: Same as on the other PR, I think the comment is wrong.

Comment (QuentinBisson) [reply helm/default-apps-gcp/templates/apps.yaml:10]: I think you can remove this comment tbh

Comment (QuentinBisson) [reply helm/default-apps-gcp/templates/apps.yaml:10]: Or say that cluster operator will ensure apps that this one depends on will be installed first

**Pull request merged**